pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -1577,6 +1577,18 @@ func (og *operationGenerator) GenerateExpandVolumeFunc(
 func (og *operationGenerator) GenerateExpandVolumeFSWithoutUnmountingFunc(
 	volumeToMount VolumeToMount,
 	actualStateOfWorld ActualStateOfWorldMounterUpdater) (volumetypes.GeneratedOperations, error) {
+	// Need to translate the spec here if the plugin is migrated so that the metrics
+	// emitted show the correct (migrated) plugin
+	if useCSIPlugin(og.volumePluginMgr, volumeToMount.VolumeSpec) {
+		csiSpec, err := translateSpec(volumeToMount.VolumeSpec)
+		if err == nil {
+			volumeToMount.VolumeSpec = csiSpec
+		}
+		// If we have an error here we ignore it, the metric emitted will then be for the
```

Comment:
This is a little dangerous to put here as it doesn't fail (or retry) on error. Could you follow the model used by `GenerateAttachVolumeFunc` where we do the translation inside the internal func itself (that get's retried) but also use this block at the end for the event recording (less critical piece). So the translation code is kind of duplicated, but the one in the critical path for actual resizing will get retried on error

Reply:
fixed